### DISCUSSION TITLE
How to "ping" a passive device with a power supply

### DISCUSSION THREAD

**Discussion by Yuke-hd** [Koenkk/zigbee2mqtt#18800] [Devices]
I've add a power supply to a battery operated "passive" [device](https://www.zigbee2mqtt.io/devices/MCCGQ11LM.html#xiaomi-mccgq11lm), making it virtually an "active" device.

The passive availability timeout (my setting is 720mins) still applies to it and the device can stay slient for longer 720mins so it gets incorrectly marked as offline

I've tried to overwrite the device type to "active" but I got the device cannot be ping error.
https://github.com/Yuke-hd/zigbee2mqtt/pull/1

My question is how to make the device report every x mins given battery capacity is a non issue in this case?

**Comment by Yuke-hd** [comment]
> Warning 2023-08-31 21:06:14Failed to ping 'test' (attempt 1/2, Read 0x00158d0008394318/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":true,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received (58)))
Warning 2023-08-31 21:06:27Failed to ping 'test' (attempt 2/2, Read 0x00158d0008394318/1 genBasic(["zclVersion"], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received (59)))

**Reply by malaakso** [reply]
You would need to modify the device firmware to keep its radio on all the time. It doesn't know that it is permanently powered, and it only wakes up to send state changes and report battery every now and then.
